crm_project_create/models/crm_lead.py

```diff
@@ -8,3 +8,10 @@ class CrmLead(models.Model):
     _inherit = "crm.lead"
 
     project_id = fields.Many2one("project.project", string="Project")
+
+    def toggle_active(self):
+        """Archive or reactivate the project and their analytic account on lead toggle."""
+        for lead in self.filtered(lambda l: l.project_id):
+            lead.sudo().project_id.active = not lead.active
+            lead.sudo().project_id.analytic_account_id.active = not lead.active
+        return super().toggle_active()
```

Comment:
Why not move `super().toggle_active()` to top and avoid the confusing `not lead.active`?

```suggestion
        res = super().toggle_active()
        for lead in self.filtered(lambda l: l.project_id):
            lead.sudo().project_id.active = lead.active
            lead.sudo().project_id.analytic_account_id.active = lead.active
        return res
```

Reply:
It makes sense. Thanks for the advice!